src/gt4py/ir/nodes.py

```diff
@@ -691,6 +691,14 @@ class AxisInterval(Node):
     end = attribute(of=AxisBound)
     loc = attribute(of=Location, optional=True)
 
+    def equals(self, other: "AxisInterval") -> bool:
```

Comment:
Consider implementing `__eq__` instead of making an `equals` method.

Reply:
I did try that first, but did not work so implemented this just get a fix posted. Can try `__eq__` again.

Reply:
Overriding `__eq__` is not possible with `attribclass`es. To exclude `loc` from the equality check one needs to add `eq=False` to the attribute, i.e.
```
loc = attribute(of=Location, optional=True, eq=False)
```
To maintain consistency of the semantics of `__eq__` with the other nodes we would also need to do this for all `loc` attributes elsewhere, which I would avoid in the dated frontend. I would propose to use a method named `is_same_interval` instead.